db/functions/base.py

```diff
@@ -0,0 +1,261 @@
+    def column(self):
+        return self.parameters[0]
+
+    @staticmethod
+    def to_sa_expression(p):
+        return column(p)
+
+
+class List(DbFunction):
+    id = 'list'
+    name = 'List'
+
+    @staticmethod
+    def to_sa_expression(*ps):
+        return list(ps)
+
+
+class Empty(DbFunction):
+    id = 'empty'
+    name = 'Empty'
+    hints = tuple([
+        hints.returns(hints.boolean),
+        hints.parameter_count(1),
+    ])
+
+    @staticmethod
+    def to_sa_expression(p):
+        return p.is_(None)
+
+
+class Not(DbFunction):
+    id = 'not'
+    name = 'Not'
+    hints = tuple([
+        hints.returns(hints.boolean),
+        hints.parameter_count(1),
+    ])
+
+    @staticmethod
+    def to_sa_expression(*p):
+        return not_(*p)
+
+
+class Equal(DbFunction):
+    id = 'equal'
+    name = 'Equal'
+    hints = tuple([
+        hints.returns(hints.boolean),
+        hints.parameter_count(2),
+    ])
+
+    @staticmethod
+    def to_sa_expression(p1, p2):
+        return p1 == p2
+
+
+class Greater(DbFunction):
+    id = 'greater'
+    name = 'Greater'
+    hints = tuple([
+        hints.returns(hints.boolean),
+        hints.parameter_count(2),
+        hints.all_parameters(hints.comparable),
+    ])
+
+    @staticmethod
+    def to_sa_expression(p1, p2):
+        return p1 > p2
+
+
+class Lesser(DbFunction):
+    id = 'lesser'
+    name = 'Lesser'
+    hints = tuple([
+        hints.returns(hints.boolean),
+        hints.parameter_count(2),
+        hints.all_parameters(hints.comparable),
+    ])
+
+    @staticmethod
+    def to_sa_expression(p1, p2):
+        return p1 < p2
+
+
+class In(DbFunction):
+    id = 'in'
+    name = 'In'
+    hints = tuple([
+        hints.returns(hints.boolean),
+        hints.parameter_count(2),
+        hints.parameter(2, hints.array),
+    ])
+
+    @staticmethod
+    def to_sa_expression(p1, p2):
+        return p1.in_(p2)
+
+
+class And(DbFunction):
+    id = 'and'
+    name = 'And'
+    hints = tuple([
+        hints.returns(hints.boolean),
+    ])
+
+    @staticmethod
+    def to_sa_expression(*ps):
+        return and_(*ps)
+
+
+class Or(DbFunction):
+    id = 'or'
+    name = 'Or'
+    hints = tuple([
+        hints.returns(hints.boolean),
+    ])
+
+    @staticmethod
+    def to_sa_expression(*ps):
+        return or_(*ps)
+
+
+class StartsWith(DbFunction):
+    id = 'starts_with'
+    name = 'Starts With'
+    hints = tuple([
+        hints.returns(hints.boolean),
+        hints.parameter_count(2),
+        hints.all_parameters(hints.string_like),
+    ])
+
+    @staticmethod
+    def to_sa_expression(p1, p2):
+        return p1.like(f'{p2}%')
+
+
+class ToLowercase(DbFunction):
+    id = 'to_lowercase'
+    name = 'To Lowercase'
+    hints = tuple([
+        hints.parameter_count(1),
+        hints.all_parameters(hints.string_like),
+    ])
+
+    @staticmethod
+    def to_sa_expression(p1):
+        return func.lower(p1)
+
+
+class ExtractURIAuthority(DbFunction):
```

Comment:
Since this depends on our custom URI type, I would not put this in the general functions namespace. I'd store it with the URI type instead.

Keep in mind that our custom types may not be installed on all Mathesar deployments.

Reply:
This should apply to any functions that depend on our custom types.